modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/RustServerCodegen.java

```diff
@@ -734,28 +731,20 @@ public CodegenOperation fromOperation(String path, String httpMethod, Operation
         }
 
         String underscoredOperationId = underscore(op.operationId);
-        op.vendorExtensions.put("operation_id", underscoredOperationId); // TODO: 5.0 Remove
         op.vendorExtensions.put("x-operation-id", underscoredOperationId);
-        op.vendorExtensions.put("uppercase_operation_id", underscoredOperationId.toUpperCase(Locale.ROOT)); // TODO: 5.0 Remove
         op.vendorExtensions.put("x-uppercase-operation-id", underscoredOperationId.toUpperCase(Locale.ROOT));
         String vendorExtensionPath = op.path.replace("{", ":").replace("}", "");
-        op.vendorExtensions.put("path", vendorExtensionPath);  // TODO: 5.0 Remove
         op.vendorExtensions.put("x-path",vendorExtensionPath);
-        op.vendorExtensions.put("PATH_ID", pathId); // TODO: 5.0 Remove
         op.vendorExtensions.put("x-path-id", pathId);
-        op.vendorExtensions.put("hasPathParams", !op.pathParams.isEmpty()); // TODO: 5.0 Remove
         op.vendorExtensions.put("x-has-path-params", !op.pathParams.isEmpty());
```

Comment:
This needs to be:

```suggestion
        op.vendorExtensions.put("x-has-path-params", hasPathParams);
```

Before we were adding hasPathParams twice, once with `!op.pathParams.isEmpty()`, and then again with `hasPathParams`. This was probably caused by a mistaken merge.

This is causing the difference in samples